### PR TITLE
wasm2c: simd support (v2)

### PR DESCRIPTION
**Pull request by wrv**
This is a properly rebased version of https://github.com/WebAssembly/wabt/pull/2021 that the maintainers can edit.

### CODE REVIEW

**Comment by keithw** [comment]
I went and rebased what I had seen as the necessary preparatory commits for this (#2080, #2081, and #2134) on top of the current main branch. (Also split out #2134 from #2081.) Once we can get those three in and the repo config updated, I think SIMD will be ready for final review and merger.

**Comment by keithw** [comment]
Urg, tried to rebase this but pushed the main branch unintentionally, which caused GitHub to auto-close it, and now I can't re-push because it's closed. Could you please push a commit here and reopen this PR?

**Comment by keithw** [comment]
Ok, here's a version rebased on the current main branch. I think this is very close. @wrv, thank you for all your work on this and for sticking with it! Some small remaining questions:

- @wrv, do you want this to go in as "Author: wrv <wrv@utexas.edu>" or do you want your (human) name on it? :-)
- Do we want/need to split out the small non-SIMD changes, in particular the below, or are these small enough that we can just piggyback them on this huge PR without worrying about it:
      - `memset(&sa , '\0', sizeof(sa));` in os_install_signal_handler / os_cleanup_signal_handler in wasm-rt-impl.c
      - printing wasm_rt_strerror(trap_code) in spec-wasm2c-prefix.c
- What are our intentions for addressing the five remaining test failures and declaring this "done"? @wrv do you want to take charge of that, or do you want to make it a first-come first-served kind of deal? And do we think we can get them fixed upstream in SIMDe or do we want to implement those routines locally for now?
- Long term, do we want to be creating a dependence on SIMDe in the generated code (and the user needing that installed to compile the wasm2c output), or do we want to be building the combined simde header file and having wasm2c splat that out at the top of the generated code?

File: test/spec-wasm2c-prefix.c
Comment (keithw) [inline test/spec-wasm2c-prefix.c:212]: Given that we never reuse f32_bits or f64_bits, maybe we should just keep these functions as-is?

File: wasm2c/wasm-rt-impl.c
Comment (keithw) [inline wasm2c/wasm-rt-impl.c:237]: This isn't strictly related to SIMD but I think it's probably harmless to do here.

**Review by keithw — APPROVED** [review]
lgtm. We will need the branch protection configuration to change before this can be merged.

**Comment by sbc100** [comment]
I think the CI tasks won't get updated except for new PRs.. feel free to land anyway.

**Comment by keithw** [comment]
Hmm, I don't think I have permissions to land it -- I'm still seeing the "asan" and "ubsan" tests as required. I think this may be a question of the branch protection rules for the "main" branch in the repo settings. (I tried opening a new PR in #2135 but it seemed the same.)

I do think we should hold off on actually landing it until @wrv gets a chance to reply and e.g. decide how he wants to sign the commit.

**Comment by sbc100** [comment]
> Hmm, I don't think I have permissions to land it -- I'm still seeing the "asan" and "ubsan" tests as required. I think this may be a question of the branch protection rules for the "main" branch in the repo settings. (I tried opening a new PR in #2135 but it seemed the same.)

The rules have been updated, but i think they don't get applied to existing PRs.

I can land this now if you like.

**Comment by sbc100** [comment]
(sorry I just saw you message to hold off.. )

**Comment by wrv** [comment]
I'm excited to see this is going in! ^.^

My human name would be good: `Author: Willy R. Vasquez <wrv@utexas.edu>`

I think a "whoever gets around to it first" approach is fine. I still haven't identified the reasons for the failing tests, but once we do we can likely upstream it in SIMDe. 

I'm leaning towards the splatting out the combined SIMDe header option, which may be easier for wasm2c users, but whatever y'all think is best.

**Comment by keithw** [comment]
Great! @sbc100 I think this is ready to land.

**Pull request merged**
